pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -154,6 +154,7 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 
 	// We use a finalizer to maintain KubeDirector state consistency;
 	// e.g. app references and ClusterStatusGens.
+	var currentHash string = calcConnectionsHash(&cr.Spec.Connections, cr.Namespace)
```

Comment:
Slightly confusing to have the comment about finalizers attached to this line here.